hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaConverter.java

```diff
@@ -73,9 +79,14 @@ Type convertType(TypeInfo typeInfo) {
             return Types.BooleanType.get();
           case BYTE:
           case SHORT:
-            throw new IllegalArgumentException("Unsupported Hive type (" +
-                ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory() +
-                ") for Iceberg tables. Consider using INT/INTEGER type instead.");
+            if (autoConvert) {
+              LOG.debug("Using auto conversion from SHORT to INTEGER");
+              return Types.IntegerType.get();
+            } else {
+              throw new IllegalArgumentException("Unsupported Hive type (" +
+                  ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory() +
+                  ") for Iceberg tables. Consider using INT/INTEGER type instead.");
+            }
```

Comment:
You could probably make this a bit cleaner by using a `Precondition`:

```java
    Preconditions.check(autoConvert,
        "Unsupported Hive type: %s, use integer instead", ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory());
    return Types.IntegerType.get();
```

That way, you'd get string formatting for free.

Reply:
The code is definitely sorter, if a little bit strange for me 😄 
Changed anyway